### PR TITLE
[WIP] Bulk update coupons [Relies on PR #1423]

### PR DESCRIPTION
**Pull request by hoffmabc**
This relies on PR #1423 getting merged first.

Here is the Bulk Update Coupons.

It accepts an array of coupon codes and replaces all coupon codes on all listings.

*POST* http://127.0.0.1:4002/ob/bulkupdatecoupons

Example Payload:
```
{"coupons": 
  [ 
    { "title": "test", "code": "test2",  "percentDiscount": 99 },
    { "title": "test", "code": "test23",  "percentDiscount": 32 } 
  ] 
}
```

### CODE REVIEW

File: core/moderation.go [hunk 1]
```diff
@@ -178,6 +178,76 @@ func (n *OpenBazaarNode) GetModeratorFee(transactionTotal uint64, paymentCoin, c
...
+			if err != nil {
+				return err
+			}
+			sl := new(pb.SignedListing)
+			err = jsonpb.UnmarshalString(string(file), sl)
+			if err != nil {
+				return err
+			}
+
+			sl.Listing.Coupons = coupons
```
Comment (placer14) [inline core/moderation.go]: Should we remove the prior coupons when we apply the new ones? Merge them?

Comment (cpacia) [reply core/moderation.go]: I think it's propbably OK to replace the existing coupons as the intent is to set all the listings to the same thing. Or so it seems to me.

Comment (hoffmabc) [reply core/moderation.go]: That's what I was thinking, but either way we have to be clear about it and make it obvious to users what's going on.

**Comment by cpacia** [comment]
Since this combines the bulk currency PR as well maybe change the title so as to reflect that this PR contains both.

**Comment by hoffmabc** [comment]
I think that's why we had the moderators setting functionality set up in this way, but @placer14 asked to re-use the UpdateListing call. I'll defer to your best judgment on this.

**Comment by cpacia** [comment]
Closing for now as this is a backlog item. Can revisit later

**Pull request closed without merging**